datafusion/sqllogictest/test_files/array.slt

```diff
@@ -2603,6 +2603,16 @@ select array_position(arrow_cast(make_array([1, 2, 3], [4, 5, 6], [5, 5, 5], [4,
 ----
 2 2
 
+query I
+SELECT array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 5)
+----
+1
+
+query I
+SELECT array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 5, 2)
```

Comment:
Can we please add a few more tests:
1. For `LargeList`
2. Error cases (e.g. for types that are still not compatible like `array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 'foo')`

Reply:
Thanks for the feedback... `LargeList` is added and works as expected. For `array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 'foo')`, this actually doesn't fail right now on this branch and returns NULL:

```
❯ SELECT array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 'foo') IS NULL;
+----------------------------------------------------------------------------------------------+
| array_position(make_array(Int64(5),Int64(2),Int64(3),Int64(4),Int64(5)),Utf8("foo")) IS NULL |
+----------------------------------------------------------------------------------------------+
| true                                                                                         |
+----------------------------------------------------------------------------------------------+
1 row in set. Query took 0.004 seconds.
```

Because this PR is leveraging the append's casting functionality, this seems to be the current behavior on main for append:

```
arrow-datafusion/datafusion-cli main ➜ ./target/debug/datafusion-cli 
DataFusion CLI v35.0.0
❯ SELECT array_append(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 'foo');
+------------------------------------------------------------------------------------+
| array_append(make_array(Int64(5),Int64(2),Int64(3),Int64(4),Int64(5)),Utf8("foo")) |
+------------------------------------------------------------------------------------+
| [5, 2, 3, 4, 5, foo]                                                               |
+------------------------------------------------------------------------------------+
1 row in set. Query took 0.028 seconds.
```

I guess as I user I'd think if the array_append case would work, I'd also expect array_position to work like that. FWIW duckdb works for both array_append and `array_position([1,2,3], 'foo')` it just returns 0 rather than NULL.

Reply:
For the second case, we return NULL which is compatible with Postgres behavior, so I think it is acceptable. You can just add the test case.

```
query I
select array_position(arrow_cast([5, 2, 3, 4, 5], 'List(Int32)'), 'foo')
----
NULL
```

Reply:
Sounds good... added in https://github.com/apache/arrow-datafusion/pull/9233/commits/4ec62896c0a87b1c20149832b68bb9757a605c77

Reply:
Thanks for adding the tests and doing the background research